src/window.rs

```diff
@@ -917,18 +917,10 @@ impl Window {
         self.window.set_cursor_position(position.into())
     }
 
-    /// Grabs the cursor, preventing it from leaving the window.
-    ///
-    /// There's no guarantee that the cursor will be hidden. You should
-    /// hide it by yourself if you want so.
-    ///
-    /// ## Platform-specific
-    ///
-    /// - **macOS:** This locks the cursor in a fixed location, which looks visually awkward.
-    /// - **iOS / Android:** Always returns an [`ExternalError::NotSupported`].
+    /// Set grabbing [mode]([`CursorGrabMode`]) on the cursor preventing it from leaving the window.
     #[inline]
-    pub fn set_cursor_grab(&self, grab: bool) -> Result<(), ExternalError> {
-        self.window.set_cursor_grab(grab)
+    pub fn set_cursor_grab_mode(&self, mode: CursorGrabMode) -> Result<(), ExternalError> {
```

Comment:
I think the old name `set_cursor_grab` is fine - will also make it easier for users to upgrade, since they will instantly see that they should use an enum instead.

Alternatively, we could call it `set_cursor_constraints` or something?

Reply:
I think keeping the old name is fine